src/Compilers/CSharp/Portable/Binder/Binder_WithExpression.cs

```diff
@@ -94,7 +94,10 @@ private BoundExpression BindWithExpression(WithExpressionSyntax syntax, Diagnost
                 }
             }
 
-            var cloneReturnType = cloneMethod?.ReturnType;
+            // Even if a clone method is not present, we should come up with
+            // a type to lookup members of the with-expression in for error recovery
+            var cloneReturnType = cloneMethod?.ReturnType ?? receiverType;
+            RoslynDebug.AssertNotNull(cloneReturnType);
```

Comment:
Why not just `Debug.Assert(cloneReturnType is object)` here? Or pick your favorite non-null check. Didn't think that we still needed to use `RoslynDebug` at this layer of the compiler.